dll/win32/mpr/wnet.c

```diff
@@ -2800,10 +2804,28 @@ DWORD WINAPI WNetDisconnectDialog1A( LPDISCDLGSTRUCTA lpConnDlgStruct )
  */
 DWORD WINAPI WNetDisconnectDialog1W( LPDISCDLGSTRUCTW lpConnDlgStruct )
 {
-    FIXME( "(%p): stub\n", lpConnDlgStruct );
-
-    SetLastError(WN_NO_NETWORK);
-    return WN_NO_NETWORK;
+    DWORD ret;
+	HRESULT hr;
+    HRESULT WINAPI SHDisconnectNetDrives(PVOID Unused);
+	
+    hr = SHDisconnectNetDrives(NULL);
+	
+	if (hr == S_OK)
+		ret = NO_ERROR;
+	else if (hr == -1)
+		ret = ERROR_CANCELLED;
+	else if (hr == ERROR_OPEN_FILES)
+		ret = ERROR_OPEN_FILES;
+	else if (hr ==  ERROR_BUSY)
+		ret = ERROR_BUSY;
+	else if (hr ==  ERROR_NO_NETWORK)
+		ret = ERROR_NO_NETWORK;
+	else if (hr ==  ERROR_NOT_ENOUGH_MEMORY)
+		ret = ERROR_NOT_ENOUGH_MEMORY;
+	else
+		ret = ERROR_EXTENDED_ERROR;
```

Comment:
I'm not sure what you're trying to achieve here?

Reply:
I've removed it all and made it so simple in the latest commit.